src/main/java/com/patchworkmc/transformer/PatchworkTransformer.java

```diff
@@ -113,16 +110,10 @@ public void accept(String name, byte[] content) {
 			accessTransformations.addFieldTransformation(holder.getField(), AccessTransformation.DEFINALIZE_MAKE_PUBLIC);
 		});
 
-		EventHandlerScanner eventHandlerScanner = new EventHandlerScanner(objectHolderScanner, eventBusSubscriber::set,
-				subscribeEvent -> {
-			subscribeEvents.add(subscribeEvent);
-
-			accessTransformations.setClassTransformation(AccessTransformation.MAKE_PUBLIC);
-
-			accessTransformations.addMethodTransformation(subscribeEvent.getMethod(), subscribeEvent.getMethodDescriptor(), AccessTransformation.MAKE_PUBLIC);
-		});
+		EventHandlerRewriter eventHandlerRewriter = new EventHandlerRewriter(objectHolderScanner, eventBusSubscriber::set,
+				subscribeEvent -> accessTransformations.setClassTransformation(AccessTransformation.MAKE_PUBLIC));
```

Comment:
Why not just use `eventHandlerRewriter.asEventSubscriber()` with `hasStaticSubscriber` or `hasInstanceSubscriber`? That would work just as well, and it means the lambda could be removed.

Reply:
I don't see an easy way of doing that without moving access transformers to their own pass on the class bytes.

Reply:
They are already applied separately, though. Mod ATs are applied after the first level transformers have been applied and the ClassNode is complete.

Reply:
hm, i'll have to double check that. PatchworkTransformer is 🍝 🖥️ so it's hard to tell